contracts/profit/src/state/buy_back.rs

```diff
@@ -33,17 +33,27 @@ pub(super) struct BuyBack {
 }
 
 impl BuyBack {
-    pub fn new(
+    pub fn try_new(
         contract_addr: Addr,
```

Comment:
[here and the member variable] rename to something more specific than just 'contract', which contract?